cli/command/service/progress/progress.go

```diff
@@ -303,7 +327,23 @@ func (u *replicatedProgressUpdater) update(service swarm.Service, tasks []swarm.
 			u.slotMap[task.Slot] = mappedSlot
 		}
 
-		if !u.done && replicas <= maxProgressBars && uint64(mappedSlot) <= replicas {
+		if !terminalState(task.DesiredState) && task.Status.State == swarm.TaskStateRunning {
+			running++
+		}
```

Comment:
Yes it would, but the current function has that side effect plus a bunch of other logic, so it's not any worse off, right?  It actually moves almost all of the progress printing to that new function. With a couple other smaller changes we could probably remove all of the progress printing from `update()`